packages/core/src/external/carequality/ihe-gateway-v2/xca/process-dr-response.ts

```diff
@@ -0,0 +1,202 @@
+    homeCommunityId: documentResponse?.HomeCommunityId,
+    repositoryUniqueId: documentResponse?.RepositoryUniqueId,
+    newDocumentUniqueId: documentResponse?.NewDocumentUniqueId,
+    newRepositoryUniqueId: documentResponse?.NewRepositoryUniqueId,
+    isNew: !fileInfo.exists,
+  };
+}
+
+async function handleSuccessResponse({
+  documentResponses,
+  outboundRequest,
+  gateway,
+}: {
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  documentResponses: any;
+  outboundRequest: OutboundDocumentRetrievalReq;
+  gateway: XCAGateway;
+}): Promise<OutboundDocumentRetrievalResp> {
+  const s3Utils = new S3Utils(region);
+
+  const idMapping = outboundRequest.documentReference.reduce(
+    (acc: Record<string, string>, entry) => {
+      if (entry.docUniqueId && entry.metriportId) {
+        acc[stripUrnPrefix(entry.docUniqueId)] = entry.metriportId;
+      }
+      return acc;
+    },
+    {}
+  );
+
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const documentReferences = Array.isArray(documentResponses)
+    ? await Promise.all(
+        documentResponses.map(async (documentResponse: DocumentResponse) =>
+          parseDocumentReference({ documentResponse, outboundRequest, s3Utils, idMapping })
+        )
+      )
+    : [
+        await parseDocumentReference({
+          documentResponse: documentResponses,
+          outboundRequest,
+          s3Utils,
+          idMapping,
+        }),
+      ];
+
+  const response: OutboundDocumentRetrievalResp = {
+    id: outboundRequest.id,
+    patientId: outboundRequest.patientId,
+    timestamp: outboundRequest.timestamp,
+    responseTimestamp: dayjs().toISOString(),
+    gateway,
+    documentReference: documentReferences,
+  };
+  return response;
+}
+
+export async function processDRResponse({
+  drResponse: { response, success, gateway, outboundRequest },
+}: {
+  drResponse: DRSamlClientResponse;
+}): Promise<OutboundDocumentRetrievalResp> {
+  if (!gateway || !outboundRequest) throw new Error("Missing gateway or outboundRequest");
+  if (success === false) {
+    return handleHTTPErrorResponse({
+      httpError: response,
+      outboundRequest,
+      gateway,
+    });
+  } else {
+    const parser = new XMLParser({
+      ignoreAttributes: false,
+      attributeNamePrefix: "_",
+      textNodeName: "_text",
+      parseAttributeValue: false,
+      removeNSPrefix: true,
+    });
+    const jsonObj = parser.parse(response);
+
+    const status = jsonObj?.Envelope?.Body?.RetrieveDocumentSetResponse?.RegistryResponse?._status
+      ?.split(":")
+      .pop();
+    const registryErrorList =
+      jsonObj?.Envelope?.Body?.RetrieveDocumentSetResponse?.RegistryResponse?.RegistryErrorList;
+    const documentResponses =
+      jsonObj?.Envelope?.Body?.RetrieveDocumentSetResponse?.DocumentResponse;
+    const soapFault = jsonObj?.Envelope?.Body?.Fault;
+
+    if ((status === "Success" || status === "PartialSuccess") && documentResponses) {
```

Comment:
we use these strings a lot - let's put em in consts in a shared location

... if they're not string literals already (ts is hard to review this way)